src/tableFormatter.ts

```diff
@@ -51,11 +51,21 @@ class MarkdownDocumentFormatter implements DocumentFormattingEditProvider {
     private detectTables(text: string) {
         const lineBreak = '\\r?\\n';
         const contentLine = '\\|?.*\\|.*\\|?';
-        const hyphenLine = '[ \\t]*\\|?( *:?-+:? *\\|)+( *:?-+:? *\\|?)[ \\t]*';
+
+        const leftSideHyphenComponent = '(?:[ \\t]*(?:\\|? *:?-+:? *\\|){1}';
```

Comment:
What is the meaning of the leading `(?:`? There seems not to be a closing `)` 🤔.

And do we need `{1}` as it can be omitted?

Reply:
Oops, the closing bracket is hiding in `leftSideHyphenComponent` 😄 I didn't chop it up properly when trying to make it more readable. You are also correct about the `{1}` - I'll make the changes. 👍